tests/README.md

```diff
@@ -1,32 +1,186 @@
+1. Add the tests to the AzureML pipeline in the corresponding [test group](./ci/azureml_tests/test_groups.py). **Please note that if you don't add your tests to the pipeline, they will not be executed.**
+
+### How to create tests for the library code
+
+You want to make sure that all your code works before you submit it to the repository. Here are guidelines for creating the unit tests:
+
+* It is better to create multiple small tests than one large test that checks all the code.
+* Use `@pytest.fixture` to create data in your tests.
+* Use the mark `@pytest.mark.gpu` if you want the test to be executed in a GPU environment. Use `@pytest.mark.spark` if you want the test to be executed in a Spark environment.
+* Use `@pytest.mark.smoke` and `@pytest.mark.integration` to mark the tests as smoke tests and integration tests.
+* Use `@pytest.mark.notebooks` if you are testing a notebook.
+* Avoid using `is` in the asserts, instead use the operator `==`.
+* Follow the pattern `assert computation == value`, for example:
+```python
+assert results["precision"] == pytest.approx(0.330753)
+```
+* Check always the limits of your computations, for example, you want to check that the RMSE between two equal vectors is 0:
+```python
+assert rmse(rating_true, rating_true) == 0
+assert rmse(rating_true, rating_pred) == pytest.approx(7.254309)
+```
+
+### How to create tests on notebooks with Papermill and Scrapbook
+
+In the notebooks of this repo, we use [Papermill](https://github.com/nteract/papermill) and [scrapbook](https://nteract-scrapbook.readthedocs.io/en/latest/) in unit, smoke and integration tests. Papermill is a tool that enables you to parameterize and execute notebooks. `scrapbook` is a library for recording a notebook’s data values and generated visual content as “scraps”. These recorded scraps can be read at a future time. We use `scrapbook` to collect the metrics in the notebooks.
+
+#### Developing unit tests with Papermill and Scrapbook
+
+Executing a notebook with Papermill is easy, this is what we mostly do in the unit tests. Next we show just one of the tests that we have in [tests/unit/examples/test_notebooks_python.py](tests/unit/examples/test_notebooks_python.py).
+
+```python
+import pytest
+import papermill as pm
+
+@pytest.mark.notebooks
+def test_sar_single_node_runs(notebooks, output_notebook, kernel_name):
+    notebook_path = notebooks["sar_single_node"]
+    pm.execute_notebook(notebook_path, output_notebook, kernel_name=kernel_name)
+```
+
+Notice that the input of the function is a fixture defined in [conftest.py](conftest.py). For more information, please see the [definition of fixtures in PyTest](https://docs.pytest.org/en/latest/fixture.html).
+
+For executing this test, first make sure you are in the correct environment as described in the [SETUP.md](../SETUP.md): 
+
+*Note that the next instruction executes the tests from the root folder.*
```

Comment:
```suggestion
*Notice that the next instruction executes the tests from the root folder.*
```
Not sure about this one, it seems note is more colloquial. If you prefer note, we should also change the one above